### ISSUE TITLE
Implement resetting to default theme state

### ISSUE THREAD

**Issue by lmorchard** [mozilla/ThemesRFun#44]
From [acceptance criteria](https://github.com/mozilla/ThemesRFun/blob/master/docs/acceptance.md):

> - [ ] It should let users reset their browser back to it's default theme state.

**Comment by lmorchard** [comment]
This could be considered done, insofar as uninstalling the add-on reverses theme changes.

Doing it while the add-on is installed might be a challenge, though - I don't think we can get a representation of the default theme to re-apply it.

**Comment by lmorchard** [comment]
Oh, actually, looks like we have [a reset API call](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/theme/reset)! One caveat:

> Note that this will always reset the theme back to the original default theme, even if the user had selected a different theme before this extension's theme was applied.

There is also [a getCurrent() API call](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/theme/getCurrent), but I think that's limited to themes using the theme API - i.e. not the built-in default / light / dark themes

**Comment by lmorchard** [comment]
@johngruen Is the little circle-arrow icon here meant to be for reset-to-default?

![capture](https://user-images.githubusercontent.com/21687/35946182-d5b5e86a-0c30-11e8-9fc3-9e088d4029d4.PNG)

**Comment by lmorchard** [comment]
One more caveat, thinking through this: If the user resets to default, we probably can't populate the color settings with that default theme. Not quite sure what to reset the editor to in this case, then.

**Comment by chuckharmston** [comment]
I'm a little concerned about the iconography here; that looks like "refresh", not "reset". Hopefully it's a non-destructive action, i.e. you can undo it?

**Comment by johngruen** [comment]
We should also make sure to have tooltips

**Comment by lmorchard** [comment]
I poked at this one a bit. @johngruen - I think we might need to do some product thinking on this.

We can revert an extension-applied theme with `browser.theme.reset()` - so that leaves us with the default theme in the browser. 

But, we can't *read* that theme. There's a `browser.theme.getCurrent()` but it returns `undefined` after a `reset()`. We can only use that API to fetch theme properties that have been set by an extension.

So, we can't set the current theme in the web editor to match the default theme after reset. And so the next edit to the theme in the editor just re-applies what was there before reset.

If you want to see what it does, I did [a quick attempt](https://github.com/lmorchard/ThemesRFun/commit/aeef68837cfba2f455940666ab20abdbcca01ce6) to wire up the reset button [in a branch](https://github.com/lmorchard/ThemesRFun/tree/44-reset-default).
